llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -1218,6 +1224,32 @@ void addInstrRequirements(const MachineInstr &MI,
       Reqs.addCapability(SPIRV::Capability::SplitBarrierINTEL);
     }
     break;
+  case SPIRV::OpSDot:
+  case SPIRV::OpUDot: {
+    if (ST.canUseExtension(SPIRV::Extension::SPV_KHR_integer_dot_product))
```

Comment:
Let's please move from the switch's case code to a separate function to make the switch code a bit cleaner.